az-core/src/main/java/azkaban/Constants.java

```diff
@@ -83,6 +83,8 @@
 
   public static final Duration MIN_FLOW_TRIGGER_WAIT_TIME = Duration.ofMinutes(1);
 
+  public static final Duration MIN_AGE_FOR_CLASSIFYING_A_FLOW_AGED = Duration.ofMinutes(20);
```

Comment:
20 minutes seems a little too high to me. Is there any empirical evidence guiding this ? (why not 5, 10 or 30 minutes, for example)

Reply:
This is to account for a sudden spike that happens at turn of the hour and day and then ramps down. This number will be made configurable because this is likely to be different. 20 seemed the best default based on the data I saw.

Reply:
Defaulting to 20 mins duration leads me to think that the metric "WEB-NumQueuedFlows" quite often during peak hours reports more than 1 queued execution for more than 20 minutes, steadily. Is this or close to this what you're seeing in our metrics?

Reply:
I think the right threshold for duration is very much related to a cluster's load and therefore IMO it should be configurable from the beginning.